src/main/webapp/js/instructorFeedbackResults.js

```diff
@@ -205,37 +205,79 @@ function updateStatsCheckBox() {
+            // and ajax_auto or ajax-response-auto class will be removed from the element.
+            if (hasAjaxAutoLoading) {
+                $ajaxAuto.click();
             }
-        }
-        var htmlString = $(e).html();
-        htmlString = htmlString.replace(expand, collapse);
-        $(e).html(htmlString);
-        var tooltipString = $(e).attr('data-original-title').replace(expand, collapse);
-        $(e).attr('title', tooltipString).tooltip('fixTitle').tooltip('show');
-    } else {
-        var j = 0;
-        for (var k = 0; k < panels.length; k++) {
-            if ($(panels[k]).attr('class').indexOf('in') !== -1) {
-                setTimeout(hideSingleCollapse, 100 * j, panels[k]);
-                j++;
+            if (hasAjaxResponseAutoLoading) {
+                $ajaxResponseAuto.click();
+            }
+            if (hasAjaxAutoLoading || hasAjaxResponseAutoLoading) {
+                // When the element needs Ajax loading, it must be collapsed because all expanded panels are
+                // already loaded. As Ajax is triggered by clicking, clicking will expand the collapsed
+                // panel.
+                continue;
             }
         }
-        var htmlStr = $(e).html();
-        htmlStr = htmlStr.replace(collapse, expand);
-        $(e).html(htmlStr);
-        var tooltipStr = $(e).attr('data-original-title').replace(collapse, expand);
-        $(e).attr('title', tooltipStr).tooltip('fixTitle').tooltip('show');
+
+        if ($(panels[idx]).attr('class').indexOf('in') === -1) {
+            // Expand the panel if the panel is collapsed at the moment
+            setTimeout(showSingleCollapse, 50 * i, panels[idx]);
+            i++;
+        }
+    }
+}
+
+function collapse(panels) {
+    var i = 0;
+    for (var inx = 0; inx < panels.length; inx++) {
+        if ($(panels[inx]).attr('class').indexOf('in') !== -1) {
+            // Collapse the panel if the panel is expanded at the moment
+            setTimeout(hideSingleCollapse, 100 * i, panels[inx]);
+            i++;
+        }
+    }
+}
+
+function updateHTMLContent(element, from, to) {
+    var htmlString = $(element).html();
+    htmlString = htmlString.replace(from, to);
+    $(element).html(htmlString);
+    var tooltipString = $(element).attr('data-original-title').replace(from, to);
+    $(element).attr('title', tooltipString).tooltip('fixTitle').tooltip('show');
+}
+
+function toggleCollapse(e, pans) {
+    var EXPAND = 'Expand';
+    var COLLAPSE = 'Collapse';
+    var panels = pans || $('div.panel-collapse');
+    var isElementAnExpandButton = $(e).html().trim().startsWith(EXPAND);
+
+    if (isElementAnExpandButton) {
+        var isElementCollapsePanelsButton = $(e).is($('#collapse-panels-button'));
+        // We might need to load {@code panels} data by Ajax if {@link #toggleCollapse} is invoked by
+        // clicking on collapse panels button.
+        var mightNeedAjaxLoading = isElementCollapsePanelsButton;
```

Comment:
If ajax loading is something to do with expanding, it can be dealt with inside the `expand` method.